src/access.d

```diff
@@ -538,3 +519,46 @@ extern (C++) bool symbolIsVisible(Scope *sc, Dsymbol s)
     case PROTpublic, PROTexport: return true;
     }
 }
+
+/**
+ * Use the most visible overload to check visibility. Later perform an access
+ * check on the resolved overload.  This function is merely a copy of
+ * overloadApply, but doesn't resolve aliases (.toAlias) because
+ * protection/visibility is an attribute of the alias not the aliasee.
+ */
+private Dsymbol mostVisibileOverload(Dsymbol s)
+{
+    if (!s.isOverloadable())
+        return s;
+
+    Dsymbol next, mostVisibile = s;
+    for (; s; s = next)
+    {
+        // alias common = mod1.templ1;
+        // alias common = mod2.templ2;
+        if (auto od = s.isOverDeclaration())
+            assert(0, "OverDeclaration");
+        // alias common = mod1.func1;
+        // alias common = mod2.func2;
+        else if (auto fa = s.isFuncAliasDeclaration())
+            assert(0, "FuncAliasDeclaration");
+        // alias name = sym;
+        // private void name(int) {}
+        else if (auto ad = s.isAliasDeclaration())
+            next = ad.overnext;
```

Comment:
Doesn't seem to work that way, seem like the function is `overloadInsert`ed into the aliasee instead.